tests/urlpatterns/tests.py

```diff
@@ -303,3 +307,21 @@ def raises_type_error(value):
             raise TypeError('This type error propagates.')
         with self.assertRaisesMessage(TypeError, 'This type error propagates.'):
             reverse('dynamic', kwargs={'value': object()})
+
+
+class DeprecationTests(TestCase):
+    @ignore_warnings(category=RemovedInDjango40Warning)
+    def test_conf_url(self):
+        urls = (
+            conf_url(r'^regex/(?P<pk>[0-9]+)/$', empty_view, name='regex'),
+            re_path(r'^regex/(?P<pk>[0-9]+)/$', empty_view, name='regex'),
+        )
+        assert str(urls[0]) == str(urls[1])
```

Comment:
Use `assertEqual()`, e.g.
```python
self.assertEqual(
    conf_url(r'^regex/(?P<pk>[0-9]+)/$', empty_view, name='regex'),
    re_path(r'^regex/(?P<pk>[0-9]+)/$', empty_view, name='regex'), 
)
```

Reply:
@felixxm `URLPattern` doesn't have an equality function. I've used `assertEqual()` but converted them to strings first. Is this ok?

`AssertionError: <URLPattern '^regex/(?P<pk>[0-9]+)/$' [name='regex']> != <URLPattern '^regex/(?P<pk>[0-9]+)/$' [name='regex']>`

Reply:
TBH, I don't think we need this test. I will chop it.